inject.sh

```diff
@@ -1,85 +1,124 @@
 #!/bin/bash
 
-pid=$(pidof "hl2_linux")
+# Verify that we are root
+if [[ ! $(id -u) -eq 0 ]]; then
+    echo "This script must be ran as root!"
+    exit 1
+fi
+
+# -s is used to only get one pid
+# If you need to get an array of processes remove it
+pid=$(pidof -s "hl2_linux")
 libpath=$(realpath "libenoch.so")
 
-if [ "$pid" == "" ]; then
-   echo "inject.sh: process not running."
+if [[ ! -f "$libpath" ]]; then
+    echo "ERROR: inject.sh: Compile cheat first!"
+    exit 1
+fi
+
+if [[ -z "$pid" ]] || [[ "$pid" == "" ]]; then
+   echo "ERROR: inject.sh: process not running."
    exit 1
 fi
 
 # Check for needed fonts. Delete these lines if you are sure all the needed
 # fonts are installed.
 cozettevector=$(fc-list | grep "CozetteVector" | wc -l)
-if [ $cozettevector -lt 1 ]; then
+if [[ $cozettevector -lt 1 ]]; then
     echo "ERROR: inject.sh: font \"CozetteVector\" not found."
     exit 1
 fi
 
+old_tmp_libpath_txt="/tmp/hl2_linux_${pid}_enoch.txt"
+old_tmp_libpath=$(head -n 1 "$old_tmp_libpath_txt")
+if [[ -z "$old_tmp_libpath" && "$old_tmp_libpath" != "" ]]; then
+    echo "inject.sh: old_tmp_libpath: $old_tmp_libpath"
+fi
+
+tmp_libpath=""
+if [[ "$1" == "unload" ]] && [[ ! -z "$2" ]]; then
+    # While unloading use the right library path
+    tmp_libpath="$2"
+else
+    # Generate new temp directory file
+    tmp_libpath="/tmp/libgl"
```

Comment:
lol 💯

Reply:
i have no idea why i wrote that yesterday

Reply:
I meant the `libgl` file name is funny. Well done.

Reply:
> I meant the `libgl` file name is funny. Well done.

~stolen from~ inspired by 😺hook